src/meili/map_matcher.cc

```diff
@@ -152,28 +169,39 @@ std::vector<MatchResult> InterpolateMeasurements(const MapMatcher& mapmatcher,
   std::vector<EdgeSegment> route = MergeRoute(mapmatcher.state_container().state(stateid),
                                               mapmatcher.state_container().state(next_stateid));
 
+  if (route.empty()) {
```

Comment:
what is this doing, what tests do we have to show that this is doing what its supposed to

Reply:
yes, after the changes. Some existing unit test in the test matcher got broken if we do not have this protection.